### PR TITLE
new(driver): support additional tracepoints into the modern BPF probe

### PR DESCRIPTION
**Pull request by Andreagit97**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area driver-modern-bpf

/area tests

**Does this PR require a change in the driver versions?**

No

**What this PR does / why we need it**:

This PR introduces 3 new tracepoints:
*  `page_fault_user`
*  `page_fault_kernel`
*  `signal_deliver`

With these new tracepoints, we should have implemented all the tracepoints supported in other drivers :) 
Since we merged the sampling logic, I've added it also to these new tracepoints
Part of our porting issue: https://github.com/falcosecurity/libs/issues/723

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by FedeDP** [comment]
/milestone next-driver

Great effort @Andreagit97 :rocket:

File: test/drivers/event_class/event_class.cpp [hunk 1]
```diff
@@ -102,29 +102,52 @@ event_test::~event_test()
...
 {
 	m_current_param = 0;
-	m_event_type = event_type;
 
-	switch(m_event_type)
+	switch(tp_code)
 	{
-	case PPME_PROCEXIT_1_E:
+	case SCHED_PROC_EXIT:
 		m_tp_set[SCHED_PROC_EXIT] = 1;
```
Comment (FedeDP) [inline test/drivers/event_class/event_class.cpp:113]:
```suggestion
		m_tp_set[tp_code] = 1;
```
and remove all first line in each switch case?

**Review by FedeDP — COMMENTED** [review]
Left a minor comment.
LGTM!

File: test/drivers/event_class/event_class.cpp (hunk 1 above)
Comment (Andreagit97) [reply test/drivers/event_class/event_class.cpp]: ty

**Review by FedeDP — APPROVED** [review]
/approve

**Review by jasondellaluce — APPROVED** [review]
/approve

**Pull request merged**